python-313/typing/readonly.py

```diff
@@ -0,0 +1,48 @@
+"""Demonstration of PEP 705: TypedDict: read-only items
+
+Use PyLance in VS Code by setting Python › Analysis: Type Checking Mode or run
+the Pyright CLI:
+
+    $ python -m pip install pyright $ pyright --pythonversion 3.13 .
+
+Extension of TypedDict:
+https://realpython.com/python38-new-features/#more-precise-types
+"""
+
+from typing import NotRequired, ReadOnly, TypedDict
+
+# class Version(TypedDict):
+#     version: str
+#     release_year: NotRequired[int | None]
+
+
+# class PythonVersion(TypedDict):
+#     version: str
+#     release_year: int
+
+
+class Version(TypedDict):
+    version: ReadOnly[str]
+    release_year: ReadOnly[NotRequired[int | None]]
```

Comment:
Does the order matter, i.e., do you know if these two are equivalent?

```python
ReadOnly[NotRequired[int | None]]
NotRequired[ReadOnly[int | None]]
```

Reply:
Yes, according to the PEP, they can be nested in any order: https://peps.python.org/pep-0705/#interaction-with-other-special-types

This works for me in VS code/Pyright as well.